src/tools/ResultsComparer/Program.cs

```diff
@@ -0,0 +1,183 @@
+
+            var benchmarkIdToDiffResults = diffResults.SelectMany(result => result.Benchmarks).ToDictionary(benchmarkResult => benchmarkResult.FullName, benchmarkResult => benchmarkResult);
+
+            return baseResults
+                .SelectMany(result => result.Benchmarks)
+                .ToDictionary(benchmarkResult => benchmarkResult.FullName, benchmarkResult => benchmarkResult) // we use ToDictionary to make sure the results have unique IDs
+                .Where(baseResult => benchmarkIdToDiffResults.ContainsKey(baseResult.Key))
+                .Select(baseResult => (baseResult.Key, baseResult.Value, benchmarkIdToDiffResults[baseResult.Key]));
+        }
+
+        private static void ExportToCsv((string id, Benchmark baseResult, Benchmark diffResult, EquivalenceTestConclusion conclusion)[] notSame, FileInfo csvPath)
+        {
+            if (csvPath == null)
+                return;
+
+            if (csvPath.Exists)
+                csvPath.Delete();
+
+            using (var textWriter = csvPath.CreateText())
+            {
+                foreach (var result in notSame)
+                {
+                    textWriter.WriteLine($"\"{result.id.Replace("\"", "\"\"")}\";base;{result.conclusion};{string.Join(';', result.baseResult.GetOriginalValues())}");
+                    textWriter.WriteLine($"\"{result.id.Replace("\"", "\"\"")}\";diff;{result.conclusion};{string.Join(';', result.diffResult.GetOriginalValues())}");
+                }
+            }
+
+            Console.WriteLine($"CSV results exported to {csvPath.FullName}");
+        }
+
+        private static string[] GetFilesToParse(string path)
+        {
+            if (Directory.Exists(path))
+                return Directory.GetFiles(path, $"*{FullBdnJsonFileExtension}", SearchOption.AllDirectories);
+            else if (File.Exists(path) || !path.EndsWith(FullBdnJsonFileExtension))
+                return new[] { path };
+            else
+                throw new FileNotFoundException($"Provided path does NOT exist or is not a {path} file", path);
+        }
+
+        // code and magic values taken from BenchmarkDotNet.Analysers.MultimodalDistributionAnalyzer
+        // See http://www.brendangregg.com/FrequencyTrails/modes.html
+        private static string GetModalInfo(Benchmark benchmark)
+        {
+            if (benchmark.Statistics.N < 12) // not enough data to tell
+                return null;
+
+            double mValue = MathHelper.CalculateMValue(new BenchmarkDotNet.Mathematics.Statistics(benchmark.GetOriginalValues()));
+            if (mValue > 4.2)
+                return "multimodal";
+            else if (mValue > 3.2)
+                return "bimodal";
+            else if (mValue > 2.8)
+                return "can have several modes";
+
+            return null;
+        }
+
+        private static double GetRatio(EquivalenceTestConclusion conclusion, Benchmark baseResult, Benchmark diffResult)
```

Comment:
I think my comment got lost in the refactor - why are we reporting straight ratios instead of relative percentages? Typically I expect to see this calculation look something like (diff-base)/base (in the case where a higher number is better.) I'd also like to see slower (regressions) represented as negative deltas instead of reversing the calculation as is done here.

Reply:
I think that base/diff, diff/base, and diff-base)/base is very subjective. For example: your preferences are different than @AndyAyersMS which are also different than @stephentoub ;)

What I have learned in BDN is that users want to customize everything, but it typically adds too much complexity to the code.

For example here to keep everyone happy I would need to introduce a new console argument, add docs for it and handle all cases in sorting the results, formatting them and aligning in the table. I don't have time for it, but I would be happy to review a PR if somebody is willing to implement it.

Reply:
I don't think it's that subjective. Using straight ratios leads to weird things sometimes - going from 10 to 8 by your method would show as 1.25, but I would think of as a -20% regression from 10. Reversing the terms so that smaller ratios are "worse", always dividing by the base, results in 0.8, which is also a non-obvious way to present the data but closer to something that makes sense in terms of how a data point relates to the previous data point. It gets weird in another way when you invert and higher numbers are worse - imagine a working set measurement going from 1235 pages loaded to 2342 pages. Ratio would tell us it is 1.9, while it is a -47% regression.

For this very specific purpose it may not matter much, but I think there is value in a consistent method for reporting data like this which makes sense across contexts. Given that this tool is not likely to be used for things other than manual investigations we can let it lie, but I'd like to take this back up in a more global sense as we move forward.

Reply:
Sounds like I will have to polish up my newsletter explaining all the ways ratios are vastly superior to other comparative measures... 

More importantly, though: I always prefer to see things reported as diff/base, whether as a percentage or ratio or whatnot, so a single column sort can order things and we can plot distributions without having to do extra math.

Reply:
@billwert Different developers have different preferences when it comes down to how their data is represented. In the past developers have asked for `base/diff`, `diff/base`, `diff-base/base`, etc. We should just make sure that we are consistent and provide transparent w.r.t. the way output data is presented.

Reply:
@AndyAyersMS I would like to read that newsletter. (Though I don't quite understand how ratios achieve sorting in ways that deltas do not.)

@jorive 
> We should just make sure that we are consistent and provide transparent w.r.t. the way output data is presented.

Agreed. I'm trying to identify which way that should go. :)